Distribution Fails to Run

### Expected Behavior

```console
.\gradlew assemble
unzip .\build\distribrutions\grails-7.0.0-SNAPSHOT.zip
.\build\distributions\grails-7.0.0-SNAPSHOT\bin\grails -version
| Grails Version: 7.0.0-SNAPSHOT
| JVM Version: 17.0.12
```

### Actual Behaviour

```console
.\gradlew assemble
unzip .\build\distribrutions\grails-7.0.0-SNAPSHOT.zip
.\build\distributions\grails-7.0.0-SNAPSHOT\bin\grails -version

Exception: java.lang.NoSuchMethodError thrown from the UncaughtExceptionHandler in thread "main"
```

### Steps To Reproduce

_No response_

### Environment Information

- Operating system: Windows 11
- JDK Version: OpenJDK 64-Bit Server VM Temurin-17.0.12+7 (build 17.0.12+7, mixed mode, sharing)

### Example Application

_No response_

### Version

7.0.0-SNAPSHOT

https://github.com/grails/grails-core/issues/13816 and https://github.com/grails/grails-core/pull/13801 got us past the initial issues.

now need to have `org.gradle:gradle-tooling-api` in the distribution

```
grails
Error: Unable to initialize main class org.grails.cli.GrailsCli
Caused by: java.lang.NoClassDefFoundError: org/gradle/tooling/BuildCancelledException
```


https://github.com/grails/grails-core/pull/13762/files set it to compileOnly so that the extra gradle repo was not needed in every project.

--- a/grails-shell/src/main/groovy/org/grails/cli/boot/GrailsDependencyVersions.groovy
+++ b/grails-shell/src/main/groovy/org/grails/cli/boot/GrailsDependencyVersions.groovy
@@ -1,5 +1,5 @@
 /*
- * Copyright 2014 original authors
+ * Copyright 2014-2024 original authors
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -96,12 +96,9 @@ class GrailsDependencyVersions implements DependencyManagement {
      *
      * @return the version with lookup from properties when required
      */
-    String versionLookup(String version){
-        if (version?.startsWith('${') && version?.endsWith('}')) {
-            return versionProperties.get(version.substring(2, version.length()-1))
-        }
-
-        return version
+    String versionLookup(String version) {
+        version?.startsWith('${') && version?.endsWith('}') ?
+                versionProperties[version[2..-2]] : version
     }
 
     protected void addDependency(String group, String artifactId, String version) {

--- a/grails-shell/src/main/groovy/org/grails/cli/boot/GrailsDependencyVersions.groovy
+++ b/grails-shell/src/main/groovy/org/grails/cli/boot/GrailsDependencyVersions.groovy
@@ -68,10 +68,40 @@ class GrailsDependencyVersions implements DependencyManagement {
 
     @CompileDynamic
     void addDependencyManagement(GPathResult pom) {
-        pom.dependencyManagement.dependencies.dependency.each { dep ->
-            addDependency(dep.groupId.text(), dep.artifactId.text(), dep.version.text())
-        }
         versionProperties = pom.properties.'*'.collectEntries { [(it.name()): it.text()] }
+        pom.dependencyManagement.dependencies.dependency.each { dep ->
+            addDependency(dep.groupId.text(), dep.artifactId.text(), versionLookup(dep.version.text()))
+        }
+    }
+
+    /**
+     * Handles properties version lookup in grails-bom
+     *
+     *   <properties>
+     *    <ant.version>1.10.15</ant.version>
+     *   </properties>
+     *
+     *   <dependencyManagement>
+     *    <dependencies>
+     *     <dependency>
+     *      <groupId>org.apache.ant</groupId>
+     *      <artifactId>ant</artifactId>
+     *      <version>${ant.version}</version>
+     *     </dependency>
+     *    </dependencies>
+     *   </dependencyManagement>
+     *
+     * @param version
+     *            either the version or the version to lookup
+     *
+     * @return the version with lookup from properties when required
+     */
+    String versionLookup(String version){
+        if (version?.startsWith('${') && version?.endsWith('}')) {
+            return versionProperties.get(version.substring(2, version.length()-1))
+        }
+
+        return version
     }
 
     protected void addDependency(String group, String artifactId, String version) {

